JPetMC.cpp

```diff
@@ -74,13 +76,21 @@ int main (int argc, char** argv)
     delete ui;
   }
 
-  if (EventMessenger::GetEventMessenger()->SetRandomSeed()) {
```

Comment:
I agree that `setRandomSeed()` should be called by default, but not in such a hard-coded way. Perhaps you could reverse the logic of the corresponding UI flag to be `useFixedSeed` (dafaulting to false) and call `setRandomSeed()` only if that was not set to true?

Reply:
Sadly I cannot. This is the same problem as we had previously, any user interaction happens at the line 71 (UImanager->ApplyCommand(command + fileName);). Not only we are setting all flags but also running the simulation.

Therefore we either setting random seed at the beginning (as I'm doing now) regardless of user flags or after setting user flags, therefore after running simulation (as it was done previously). Ability to set those flags and actually control random engine requires changing logic and executing that function at the level different that main function

Reply:
I think I understand now. Then we probably should do it through a respective class (RunManager perhaps?) and its respective messenger.